### PR TITLE
Add sample app

### PR DESCRIPTION
**Pull request by svrnm**
A sample app for our instrumentation, still some work that needs to be done, but it "works":

Run the individual apps:
* Frontend: `npm install ; node index.js` (should start on port 3000)
* Backend1: `flask run` (should start on port 5000)
* Backend2: `dotnet watch` (should start on port 5165)

Send traffic to the frontend with `curl localhost:3000`

### CODE REVIEW

File: app/backend2/obj/backend2.csproj.nuget.dgspec.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
+  "format": 1,
+  "restore": {
+    "/Users/neumanns/Projects/kubecon-eu-2023-opentelemetry-kubernetes-tutorial/app/backend2/backend2.csproj": {}
```
Comment (frzifus) [inline app/backend2/obj/backend2.csproj.nuget.dgspec.json:4]: Is this supposed to be committed?

**Comment by frzifus** [comment]
hm.. backend2 seems to have some issues:
```
Unhandled exception. System.IO.IOException: The configured user limit (128) on the number of inotify instances has been reached, or the per-process limit on the number of open file descriptors has been reached.
   at System.IO.FileSystemWatcher.StartRaisingEvents()
   at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.TryEnableFileSystemWatcher()
   at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.CreateFileChangeToken(String filter)
   at Microsoft.Extensions.Primitives.ChangeToken.OnChange(Func`1 changeTokenProducer, Action changeTokenConsumer)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider..ctor(FileConfigurationSource source)
   at Microsoft.Extensions.Configuration.Json.JsonConfigurationSource.Build(IConfigurationBuilder builder)
   at Microsoft.Extensions.Configuration.ConfigurationManager.AddSource(IConfigurationSource source)
   at Microsoft.Extensions.Configuration.ConfigurationManager.Microsoft.Extensions.Configuration.IConfigurationBuilder.Add(IConfigurationSource source)
   at Microsoft.Extensions.Hosting.HostingHostBuilderExtensions.ApplyDefaultAppConfiguration(HostBuilderContext hostingContext, IConfigurationBuilder appConfigBuilder, String[] args)
   at Microsoft.Extensions.Hosting.HostApplicationBuilder..ctor(HostApplicationBuilderSettings settings)
   at Microsoft.AspNetCore.Builder.WebApplicationBuilder..ctor(WebApplicationOptions options, Action`1 configureDefaults)
   at Microsoft.AspNetCore.Builder.WebApplication.CreateBuilder(String[] args)
   at Program.<Main>$(String[] args) in /App/Program.cs:line 1
```

**Comment by frzifus** [comment]
leading to :smile:  
```
$ k logs -n demo frontend-deployment-767f4b49cc-plgx4
Example app listening on port 3000
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: connect ECONNREFUSED 10.43.170.234:5165
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16)
Emitted 'error' event on ClientRequest instance at:
    at Socket.socketErrorListener (node:_http_client:502:9)
    at Socket.emit (node:events:513:28)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '10.43.170.234',
  port: 5165
}

Node.js v18.15.0
```

**Comment by frzifus** [comment]
But we can point both things to backend1 :1st_place_medal: 

https://frontend-5c209bc019fb6c00c896c6d075.klimlive.de/

File: app/backend2/obj/backend2.csproj.nuget.dgspec.json (hunk 1 above)
Comment (svrnm) [reply app/backend2/obj/backend2.csproj.nuget.dgspec.json:4]: I don't know? I have no idea what I am doing with dotnet :D

**Comment by svrnm** [comment]
> Added k8s manifests directly to your branch - hope thats ok ^^ - https://github.com/pavolloffay/kubecon-eu-2023-opentelemetry-kubernetes-tutorial/commit/0f3bfb2fe729891051b8b515157e6f356f63f785

YES! Thanks

**Comment by svrnm** [comment]
So, I improved a few things
* add some stuff to gitignore
* (hopefully) fixed the .NET application, set the port consistently to 5165
* added a loadgenerator

**Comment by svrnm** [comment]
* added a loadgen deployment

**Comment by svrnm** [comment]
* added us as potential players (so we can use that later for sensitive data filtering)
* one of us is a cheater (guess who;-) but everytime a 7 or 8 is rolled, a delay is introduced

File: app/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Sample Application
```
Comment (pavolloffay) [inline app/README.md:1]: @svrnm could you please provide a couple of instructions on how to run it? 

e..g via the docker-compose and/or k8s?

File: app/k8s.yaml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  template:
+    metadata:
+      labels:
+        app: backend1
+      annotations:
+        instrumentation.opentelemetry.io/inject-python: "true"
+    spec:
+      containers:
+      - name: backend1
+        image: ghcr.io/frzifus/kubecon-2023-backend1:latest
```
Comment (pavolloffay) [inline app/k8s.yaml:22]: It would be good to setup publishing of images in this repo

**Review by pavolloffay — APPROVED** [review]
I will merge it, the review comments can be fixed in a follow up PRs

**Pull request merged**